src/components/tab-bar/tab-bar.tsx

```diff
@@ -66,6 +66,10 @@ export class TabBar {
 
     // eslint-disable-next-line @stencil/own-methods-must-be-private
     public componentDidUnload() {
+        if (!this.setupMdc) {
```

Comment:
This will almost always be `false` so by having this condition we will never run the teardown when the component is destroyed.

It would be better to have checks for `this.scrollArea` and `this.mdcTabBar` in the `teardown` method.